core/src/main/java/org/infinispan/persistence/manager/PersistenceManager.java

```diff
@@ -222,4 +255,28 @@ void prepareAllTxStores(Transaction transaction, BatchModification batchModifica
     * @return true if all configured stores are available and ready for read/write operations.
     */
    boolean isAvailable();
+
+   /**
+    * Notifies any underlying segmented stores that the segments provided are owned by this cache and to start/configure
+    * any underlying resources required to handle requests for entries on the given segments.
+    * <p>
+    * This only affects stores that are not shared as shared stores have to keep all segments running at all times
+    * @param segments segments this cache owns
+    * @return if a configured store couldn't add segments (non segmented and non shared)
```

Comment:
It seems strange to me that we return true if segments can't be added.

Reply:
The problem is normally this method would say true if the segments were added (the issue is there could be multiple stores). I can change it instead to return false if not all stores could add the given segment.